cpp/src/io/parquet/reader_impl_helpers.hpp

```diff
@@ -234,6 +234,55 @@ class aggregate_reader_metadata {
 
   [[nodiscard]] auto get_num_row_groups() const { return num_row_groups; }
 
+  /**
+   * @brief Checks if a schema index from 0th source is mapped to the specified file index
+   *
+   * @return True if schema index is mapped
+   */
+  [[nodiscard]] inline bool is_schema_index_mapped(int schema_idx, int pfm_idx) const
+  {
+    // Check if schema_idx or pfm_idx is invalid
+    if (schema_idx < 0 or pfm_idx < 0 or pfm_idx >= static_cast<int>(per_file_metadata.size())) {
+      return false;
+    }
+
+    // True if root index requested or zeroth file index or schema_idx maps doesn't exist. (i.e.
+    // schemas are identical).
+    if (schema_idx == 0 or pfm_idx == 0 or schema_idx_maps.empty()) { return true; }
+
+    // Check if mapped
+    auto const& schema_idx_map = schema_idx_maps[pfm_idx - 1];
+    return schema_idx_map.find(schema_idx) != schema_idx_map.end();
+  }
+
+  /**
+   * @brief Maps schema index from 0th source file to the specified file index
+   *
+   * @return Mapped schema index
+   */
+  [[nodiscard]] inline int map_schema_index(int schema_idx, int pfm_idx) const
+  {
+    // Check if schema_idx or pfm_idx is invalid
+    CUDF_EXPECTS(
+      schema_idx >= 0 and pfm_idx >= 0 and pfm_idx < static_cast<int>(per_file_metadata.size()),
+      "Parquet reader encountered an invalid schema_idx or pfm_idx",
+      std::invalid_argument);
```

Comment:
not sure if we should throw invalid_argument in internal functions. IMO it only applies to user-provided arguments.

Reply:
I guess `std::out_of_range` would be better here.